### PR TITLE
IGAPP-1079: Ltr content rtl layouts

### PR DESCRIPTION
**Pull request by f1sh1918**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

1. Go to Testumgebung -> switch lang to AR -> check  local news
2. go to events and check list items (details are remote content)

Note: 

- Couldn't find anything to do for offers (mentioned in the ticket)
- We can think about using the new text component everywhere (separate refactoring ticket)

### CODE REVIEW

File: native/src/components/Caption.tsx [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import React, { ReactElement } from 'react'
 import styled from 'styled-components/native'
 
-const H1 = styled.Text`
+import Text from './base/Text'
```
Comment (steffenkleinle) [inline native/src/components/Caption.tsx:4]: Why `base` subfolder?

File: native/src/components/base/Text.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { TextProps, Text as RNText } from 'react-native'
+import styled from 'styled-components/native'
+
+import { contentAlignmentRTLText } from '../../constants/contentDirection'
+
+const StyledText = styled(RNText)<{ alignment: 'left' | 'right' }>`
+  text-align: ${props => props.alignment};
+`
+/** Direction aware text component */
+const Text = (props: TextProps): ReactElement => {
```
Comment (steffenkleinle) [inline native/src/components/base/Text.tsx:11]: :+1:

File: native/src/constants/contentDirection.ts [hunk 3]
```diff
@@ -9,3 +9,15 @@ export const contentDirection = (contentLanguage: string): 'row' | 'row-reverse'
   isContentDirectionReversalRequired(contentLanguage) ? 'row-reverse' : 'row'
 export const contentAlignment = (contentLanguage: string): 'right' | 'left' =>
   isContentDirectionReversalRequired(contentLanguage) ? 'right' : 'left'
+export const isRTLText = (text: string): boolean => {
+  const rtlCount = (text.match(/[\u0591-\u07FF\uFB1D-\uFDFD\uFE70-\uFEFC]/g) || []).length
+  // eslint-disable-next-line no-misleading-character-class
```
Comment (steffenkleinle) [inline native/src/constants/contentDirection.ts:14]:
```suggestion
```

File: native/src/constants/contentDirection.ts [hunk 4]
```diff
@@ -9,3 +9,15 @@ export const contentDirection = (contentLanguage: string): 'row' | 'row-reverse'
...
+  const ltrCount = (
+    text.match(
+      // eslint-disable-next-line no-misleading-character-class
```
Comment (steffenkleinle) [inline native/src/constants/contentDirection.ts:17]: Why is this "misleading"?

File: native/src/constants/contentDirection.ts [hunk 5]
```diff
@@ -9,3 +9,15 @@ export const contentDirection = (contentLanguage: string): 'row' | 'row-reverse'
   isContentDirectionReversalRequired(contentLanguage) ? 'row-reverse' : 'row'
 export const contentAlignment = (contentLanguage: string): 'right' | 'left' =>
   isContentDirectionReversalRequired(contentLanguage) ? 'right' : 'left'
+export const isRTLText = (text: string): boolean => {
```
Comment (steffenkleinle) [inline native/src/constants/contentDirection.ts:12]: Could you please add some tests for that?

File: native/src/constants/contentDirection.ts [hunk 6]
```diff
@@ -9,3 +9,15 @@ export const contentDirection = (contentLanguage: string): 'row' | 'row-reverse'
...
+  const rtlCount = (text.match(/[\u0591-\u07FF\uFB1D-\uFDFD\uFE70-\uFEFC]/g) || []).length
+  // eslint-disable-next-line no-misleading-character-class
+  const ltrCount = (
+    text.match(
+      // eslint-disable-next-line no-misleading-character-class
+      /[A-Za-z\u00C0-\u00C0\u00D8-\u00F6\u00F8-\u02B8\u0300-\u0590\u0800-\u1FFF\u2C00-\uFB1C\uFDFE-\uFE6F\uFEFD-\uFFFF]/g
```
Comment (steffenkleinle) [inline native/src/constants/contentDirection.ts:18]: Not sure if makes sense to add some comments explaining the regexs.

**Review by steffenkleinle — APPROVED** [review]
Tested on android, Works fine for events and news.

**Comment by steffenkleinle** [comment]
> This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/). You can find it there by looking for an issue with the key which is mentioned in the title of this pull request. It starts with the keyword **IGAPP**.
> 
>     1. Go to Testumgebung -> switch lang to AR -> check  local news
> 
>     2. go to events and check list items (details are remote content)
> 
> 
> Note:
> 
>     * Couldn't find anything to do for offers (mentioned in the ticket)

I don't understand that and can't find anything in the ticket.
 
>     * We can think about using the new text component everywhere (separate refactoring ticket)

:+1:

File: native/src/components/Caption.tsx (hunk 1 above)
Comment (f1sh1918) [reply native/src/components/Caption.tsx:4]: i prefer to split up components a little bit and here we are overwriting ReactNative base components. I think we lose overview with this huge amount of components.

File: native/src/constants/contentDirection.ts (hunk 4 above)
Comment (f1sh1918) [reply native/src/constants/contentDirection.ts:17]: don't ask me i checked the linting rule.. probably due to unicode chars?
https://eslint.org/docs/latest/rules/no-misleading-character-class

File: native/src/constants/contentDirection.ts (hunk 5 above)
Comment (f1sh1918) [reply native/src/constants/contentDirection.ts:12]: check last commit

File: native/src/constants/contentDirection.ts (hunk 4 above)
Comment (SteffiStuffel) [reply native/src/constants/contentDirection.ts:17]: Wouldn't it be fine, to just have [a-z] and [A-Z]?

**Review by SteffiStuffel — APPROVED** [review]
Not tested.

File: native/src/constants/contentDirection.ts (hunk 6 above)
Comment (steffenkleinle) [reply native/src/constants/contentDirection.ts:18]: or at least where its from

File: native/src/constants/contentDirection.ts (hunk 4 above)
Comment (f1sh1918) [reply native/src/constants/contentDirection.ts:17]: well there are `ö`and `ä` and some nordic chars and other stuff. i think its no fault to have it more exact.
Here is the bug for this ESLint Rule. It should not mark thus...
https://github.com/eslint/eslint/issues/15080
Doesn't look like it gonna be fixed soon

File: native/src/constants/contentDirection.ts (hunk 6 above)
Comment (f1sh1918) [reply native/src/constants/contentDirection.ts:18]: well okay i mean this rule checks for unicode chars of ltr languages and a-z but i can add a comment

File: native/src/constants/contentDirection.ts (hunk 3 above)
Comment (f1sh1918) [reply native/src/constants/contentDirection.ts]: removed

**Pull request merged**